online-shop/src/app/login/login.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ActivatedRoute, Router } from '@angular/router';
+import { User } from './user';
+import { FormBuilder, FormControl } from '@angular/forms';
+import { Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-login-form',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css'],
+})
+export class LoginComponent implements OnInit {
+
+    user : User = {username:'', password:'', fullName:'',roles:['']};
+  
+
+  profileForm = this.fb.group({
+    username: ['', [Validators.required, Validators.minLength(2)]],
+    password: ['', [Validators.required, Validators.nullValidator]],
+  });
+
+  constructor(
+    private http: HttpClient,
+    private fb: FormBuilder,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    
+  }
+
+  submit(): void {
+    this.user.username = this.profileForm.value.username || '{}';
```

Comment:
**question:** why do you have `|| '{}'` that is neither an empty object nor an empty string